src/averageAges.js

```diff
@@ -1,58 +1,42 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menFromThePeople = people.filter(person => century
+    ? Math.ceil(person.died / 100) === century && person.sex === 'm'
+    : person.sex === 'm'
+  );
+
+  const mensAverageAge = menFromThePeople.reduce(
+    (acc, person) =>
+      acc + (person.died - person.born), 0) / menFromThePeople.length;
+
+  return mensAverageAge;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average ave of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenFromThePeople = people.filter(person => withChildren
+    ? people.some(children => children.mother === person.name)
+    : person.sex === 'f'
+  );
+
+  const womenAverageAge = womenFromThePeople.reduce((accumulator, person) =>
+    accumulator + (person.died - person.born), 0) / womenFromThePeople.length;
+
+  return womenAverageAge;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a mother and her
- * child in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for mothers who have son.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const ages = [];
+
+  people.map(mother => {
```

Comment:
класно закрутив!)
тільки момент що ти використав map але не використовуєш масив який він передає, що не дуже правильно.
Тому тут можна замінити на .forEach і має працювати так само..
```suggestion
  people.forEach (mother => {
```

Reply:
Спасибо, ты прав. Нужно чуть больше практики, чтобы понять, где и какой метод лучше